### ISSUE TITLE
conflit between guzzlehttp/oauth-subscriber & guzzlehttp/psr7 

### ISSUE THREAD

**Issue by benji07** [mautic/mautic#11175]
### Mautic Version

4.2.x series

### PHP version

7.4

### What browsers are you seeing the problem on?

Not relevant

### What happened?

When installing the latest version of mautic (i just clone the branch 4.x), composer install the following version

> Installing guzzlehttp/psr7 (2.2.1): Extracting archive
> Installing guzzlehttp/oauth-subscriber (0.5.0): Extracting archive

But inside `guzzlehttp/oauth-subscriber` they still use code that were remove in `guzzlehttp/psr7` version 2.0.

The function `\GuzzleHttp\Psr7\parse_query` was deprecated a long time ago, and they publish a v0.6.0 of `guzzlehttp/oauth-subscriber` to add the support of psr7 v2.x.

How can I submit a PR to upgrade the package to ^0.6.0 instead of ^0.5.0 ?

### How can we reproduce this issue?

Step 1: Install mautic by cloning it (and staying on the 4.x branch)
Step 2: `composer install`

Step 3: try to create an HttpClient that use OAuth1 (i changed the content of my index_dev.php with the following code)

```
$stack = \GuzzleHttp\HandlerStack::create();

$middleware = new \GuzzleHttp\Subscriber\Oauth\Oauth1([
    'consumer_key'    => 'consumer-key',
    'consumer_secret' => 'consumer-secret',
]);
$stack->push($middleware);

$httpClient = new \GuzzleHttp\Client([
    'base_uri' => 'http://localhost/',
    'handler' => $stack,
    'auth' => 'oauth',
]);

$httpClient->get('/');
```

Step 4: Run the code `php index_dev.php`

### Relevant log output

```shell
Fatal error: Uncaught Error: Call to undefined function GuzzleHttp\Psr7\parse_query() in /Users/benjamin/Sites/test/vendor/guzzlehttp/oauth-subscriber/src/Oauth1.php:149
Stack trace:
#0 /Users/benjamin/Sites/test/vendor/guzzlehttp/oauth-subscriber/src/Oauth1.php(102): GuzzleHttp\Subscriber\Oauth\Oauth1->getSignature(Object(GuzzleHttp\Psr7\Request), Array)
#1 /Users/benjamin/Sites/test/vendor/guzzlehttp/oauth-subscriber/src/Oauth1.php(88): GuzzleHttp\Subscriber\Oauth\Oauth1->onBefore(Object(GuzzleHttp\Psr7\Request))
#2 /Users/benjamin/Sites/test/vendor/guzzlehttp/guzzle/src/PrepareBodyMiddleware.php(35): GuzzleHttp\Subscriber\Oauth\Oauth1->GuzzleHttp\Subscriber\Oauth\{closure}(Object(GuzzleHttp\Psr7\Request), Array)
#3 /Users/benjamin/Sites/test/vendor/guzzlehttp/guzzle/src/Middleware.php(31): GuzzleHttp\PrepareBodyMiddleware->__invoke(Object(GuzzleHttp\Psr7\Request), Array)
#4 /Users/benjamin/Sites/test/vendor/guzzlehttp/guzzle/src/RedirectMiddleware.php(71): GuzzleHttp\Middleware::GuzzleHttp\{closure}(Object(GuzzleHttp\Psr7\Request), Array)
#5 /Users/benjamin/Sites/test/vendor/guzzlehttp/guzzle/src/Middleware.php(63): GuzzleHttp\RedirectMiddleware->__invoke(Object(GuzzleHttp\Psr7\Request), Array)
#6 /Users/benjamin/Sites/test/vendor/guzzlehttp/guzzle/src/HandlerStack.php(75): GuzzleHttp\Middleware::GuzzleHttp\{closure}(Object(GuzzleHttp\Psr7\Request), Array)
#7 /Users/benjamin/Sites/test/vendor/guzzlehttp/guzzle/src/Client.php(331): GuzzleHttp\HandlerStack->__invoke(Object(GuzzleHttp\Psr7\Request), Array)
#8 /Users/benjamin/Sites/test/vendor/guzzlehttp/guzzle/src/Client.php(168): GuzzleHttp\Client->transfer(Object(GuzzleHttp\Psr7\Request), Array)
#9 /Users/benjamin/Sites/test/vendor/guzzlehttp/guzzle/src/Client.php(187): GuzzleHttp\Client->requestAsync('GET', Object(GuzzleHttp\Psr7\Uri), Array)
#10 /Users/benjamin/Sites/test/vendor/guzzlehttp/guzzle/src/ClientTrait.php(44): GuzzleHttp\Client->request('GET', '/', Array)
#11 /Users/benjamin/Sites/test/index_dev.php(25): GuzzleHttp\Client->get('/')
#12 {main}
  thrown in /Users/benjamin/Sites/test/vendor/guzzlehttp/oauth-subscriber/src/Oauth1.php on line 149
```

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

**Comment by RCheesley** [comment]
Thanks for raising the bug report @benji07  - we would welcome a PR. Here is the info on how to get started: https://contribute.mautic.org/contributing-to-mautic/developer

Please make bug fixes against the 4.3 branch for the time being, unless they are a BC Break in which case, against the 5.x branch.

Shout in #t-product if you need any help at all - get an invite at https://mautic.org/slack :)

**Comment by RCheesley** [comment]
Closing this as the PR is merged! 🎉
